File: viewHoldersDemo/src/main/java/vn/tale/viewholdersdemo/glide/MyAppExtension.java
package vn.tale.viewholdersdemo.glide;

import com.bumptech.glide.annotation.GlideExtension;
import com.bumptech.glide.annotation.GlideOption;
import com.bumptech.glide.request.RequestOptions;
import vn.tale.viewholdersdemo.R;

@GlideExtension
public class MyAppExtension {
  private static final int MINI_THUMB_SIZE = 100;

  private MyAppExtension() {
  } // utility class

  @GlideOption
  public static void miniThumb(RequestOptions options) {
    options
        .fitCenter()
        .placeholder(R.drawable.ic_placeholder);
  }
}

File: sample/src/main/java/vn/tiki/sample/glide/MyAppExtension.kt
package vn.tiki.sample.glide

import com.bumptech.glide.annotation.GlideExtension
import com.bumptech.glide.annotation.GlideOption
import com.bumptech.glide.request.RequestOptions
import vn.tiki.sample.R

@GlideExtension
object MyAppExtension {

  @GlideOption
  @JvmStatic
  fun miniThumb(options: RequestOptions) {
    options
        .fitCenter()
        .placeholder(R.drawable.ic_placeholder)
  }
}// utility class